File: JS/16 - Binary Numbers and Bit Manipulation - Greg Hogg DSA Course Materials Lecture 16.js
// Binary Numbers and Bit Manipulation - Greg Hogg DSA Course Materials Lecture 16


var binaryToDecimal = function(binary) {
    return parseInt(binary, 2);
};

console.log((5).toString(2));  // Decimal to Binary

var binary_5 = '101';
console.log(binaryToDecimal(binary_5));  // Convert Binary to Decimal

console.log(5 & 7);  // Bitwise AND

console.log(5 | 7);  // Bitwise OR

console.log(5 ^ 7);  // Bitwise XOR

console.log(~5);  // Bitwise NOT

console.log(5 << 2);  // Left Shift

console.log(5 >> 1);  // Arithmetic (Signed) Right Shift

console.log((890).toString(16));  // Hexadecimal (Base 16)

File: C++/16 - Binary Numbers and Bit Manipulation - Greg Hogg DSA Course Materials Lecture 16.cpp
// Binary Numbers and Bit Manipulation - Greg Hogg DSA Course Materials Lecture 16

#include <iostream>
#include <bitset>

int main() {
    // Decimal to Binary
    std::cout << std::bitset<8>(5).to_string().substr(3) << std::endl;

    // Convert Binary to Decimal
    std::string binary_5 = "101";
    std::cout << std::stoi(binary_5, nullptr, 2) << std::endl;

    // Bitwise AND
    std::cout << (5 & 7) << std::endl;

    // Bitwise OR
    std::cout << (5 | 7) << std::endl;

    // Bitwise XOR
    std::cout << (5 ^ 7) << std::endl;

    // Bitwise NOT
    std::cout << (~5) << std::endl;

    // Left Shift
    std::cout << (5 << 2) << std::endl;

    // Arithmetic (Signed) Right Shift
    std::cout << (5 >> 1) << std::endl;

    // Hexadecimal (Base 16)
    std::cout << std::hex << 890 << std::endl;

    return 0;
}
